// @backstage/plugin-user-settings/dist/components/General/UserSettingsPinToggle.esm.js
import { jsxs, jsx } from 'react/jsx-runtime';
import ListItem from '@material-ui/core/ListItem';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import ListItemText from '@material-ui/core/ListItemText';
import Switch from '@material-ui/core/Switch';
import Tooltip from '@material-ui/core/Tooltip';
import { useSidebarPinState } from '@backstage/core-components';

const UserSettingsPinToggle = () => {
  const { isPinned, toggleSidebarPinState } = useSidebarPinState();
  return /* @__PURE__ */ jsxs(ListItem, { children: [
    /* @__PURE__ */ jsx(
      ListItemText,
      {
        primary: "Pin Sidebar",
        secondary: "Prevent the sidebar from collapsing"
      }
    ),
    /* @__PURE__ */ jsx(ListItemSecondaryAction, { children: /* @__PURE__ */ jsx(
      Tooltip,
      {
        placement: "top",
        arrow: true,
        title: `${isPinned ? "Unpin" : "Pin"} Sidebar`,
        children: /* @__PURE__ */ jsx(
          Switch,
          {
            color: "primary",
            checked: isPinned,
            onChange: () => toggleSidebarPinState(),
            name: "pin",
            inputProps: { "aria-label": "Pin Sidebar Switch" }
          }
        )
      }
    ) })
  ] });
};

export { UserSettingsPinToggle };
