// @backstage/plugin-user-settings/dist/components/AuthProviders/EmptyProviders.esm.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { EmptyState, CodeSnippet } from '@backstage/core-components';

const EXAMPLE = `auth:
  providers:
    google:
      development:
        clientId: \${AUTH_GOOGLE_CLIENT_ID}
        clientSecret: \${AUTH_GOOGLE_CLIENT_SECRET}
`;
const EmptyProviders = () => /* @__PURE__ */ jsx(
  EmptyState,
  {
    missing: "content",
    title: "No Authentication Providers",
    description: "You can add Authentication Providers to Backstage which allows you to use these providers to authenticate yourself.",
    action: /* @__PURE__ */ jsxs(Fragment, { children: [
      /* @__PURE__ */ jsxs(Typography, { variant: "body1", children: [
        "Open ",
        /* @__PURE__ */ jsx("code", { children: "app-config.yaml" }),
        " and make the changes as highlighted below:"
      ] }),
      /* @__PURE__ */ jsx(
        CodeSnippet,
        {
          text: EXAMPLE,
          language: "yaml",
          showLineNumbers: true,
          highlightedNumbers: [3, 4, 5, 6, 7, 8],
          customStyle: { background: "inherit", fontSize: "115%" }
        }
      ),
      /* @__PURE__ */ jsx(
        Button,
        {
          variant: "contained",
          color: "primary",
          href: "https://backstage.io/docs/auth/add-auth-provider",
          children: "Read More"
        }
      )
    ] })
  }
);

export { EmptyProviders };
