// @backstage/plugin-search/dist/components/SidebarSearchModal/SidebarSearchModal.esm.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import SearchIcon from '@material-ui/icons/Search';
import { SidebarItem } from '@backstage/core-components';
import { SearchModal } from '../SearchModal/SearchModal.esm.js';
import { SearchModalProvider, useSearchModal } from '../SearchModal/useSearchModal.esm.js';

const SidebarSearchModalContent = (props) => {
  const { state, toggleModal } = useSearchModal();
  const Icon = props.icon ? props.icon : SearchIcon;
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsx(
      SidebarItem,
      {
        className: "search-icon",
        icon: Icon,
        text: "Search",
        onClick: toggleModal
      }
    ),
    /* @__PURE__ */ jsx(
      SearchModal,
      {
        ...state,
        toggleModal,
        resultItemComponents: props.resultItemComponents,
        children: props.children
      }
    )
  ] });
};
const SidebarSearchModal = (props) => {
  return /* @__PURE__ */ jsx(SearchModalProvider, { children: /* @__PURE__ */ jsx(SidebarSearchModalContent, { ...props }) });
};

export { SidebarSearchModal };
